gateway-provider-security-jwt/src/main/java/org/apache/knox/gateway/provider/federation/jwt/filter/AbstractJWTFilter.java

```diff
@@ -381,11 +390,29 @@ protected boolean validateToken(final HttpServletRequest request, final HttpServ
     return false;
   }
 
-  private boolean isTokenEnabled(String tokenId) throws UnknownTokenException {
-    final TokenMetadata tokenMetadata = tokenStateService == null ? null : tokenStateService.getTokenMetadata(tokenId);
+  private boolean isTokenEnabled(TokenMetadata tokenMetadata) throws UnknownTokenException {
     return tokenMetadata == null ? true : tokenMetadata.isEnabled();
   }
 
+  private boolean isNotIdle(TokenMetadata tokenMetadata) throws UnknownTokenException {
```

Comment:
I think isNotIdleLimitExceeded(tokenMetadata) (or something similar) might be a more accurate method name. If we're getting a request, the client is not currently idle. Actually, I would prefer to avoid the negative perspective, using hasIdleLimitExpired(tokenMetadata) and then !hasIdleLimitExpired(tokenMetadata) in it use. This is all a rather small point though, and perhaps not worth worrying about.

Reply:
Changed to `isIdleTimeoutLimitNotExceeded` (I'm also in favor of using method names that do not need to negate, if possible).

Reply:
@pzampino - Did you check if the new PS satisfies what you asked here?